supplementary_style_guide/style_guidelines/general-formatting.adoc

```diff
@@ -21,3 +21,72 @@ When the format _day Month year_ causes a presentation or clarity issue, use ins
 
 [[product-names-versions-ref]]
 == Product names and version references
```

Comment:
@bergerhoffer @cbudz @inoxx03 :
This entire topic is about Asciidoc implementation guidelines, not about style guidelines. I thought that we agreed as a group that the style guide would provide Asciidoc examples, but I didn't think that entire Asciidoc implementation modules are within the scope of this guide.

Was there a decision made that I'm not aware of? Seems like this should be part of the Asciidoc Quick Reference that we talked about last week.

Reply:
@stoobie So this has been an item on the list of recommendations since day 1. I agree that it crosses a bit into implementation, as did the technology preview item. We did discuss the TP one at one of our meetings a month or two back. (it was in this list of items that weren't okayed by teams when we sent them out: https://docs.google.com/document/d/1rKTZXqjS8QJZ-gPpXp-UOFLYaX8IJKuQl2Pzxr-rQwQ/edit). The consensus there was to potentially move it to more of an implementation-specific section. The same could potentially be considered for this item.

However, @inoxx03, the guidance you're adding here goes a lot further than the initial recommendation that we sent out. This is what was in the spreadsheet for this item:

```
Define attributes for product name and product version in a common attributes file, to be included in all books/master.adoc files.

Use the product name attribute for all instances of the product name where possible. Some products define more than one attribute, for example, if there is a full name with an abbreviation, and a short name of just the abbreviation.

Use the product version attribute for instances where the product version should change with the next version and the content would still be correct. Do not use the product version attribute if the version should not change. For example, if a feature was introduced in a certain version, the version should be hard coded.
```

It's moreso general tips/suggestions for the best practice of using attributes for these items. And less about the actual implementation of what to name the file, where to put it, etc.

I'm torn on what to do about this one. I think that it's a good idea to recommend using attributes for things like product name and versions, but I'd be hesitant to include specifics without coordinating with the FCC group to make sure that what we recommend is going to work when sharing content across products.

@redhat-documentation/ccs-style-council Thoughts?

Reply:
@bergerhoffer My initial intention was to provide some context and additional information around the basic recommendations that have been noted down in the spreadsheet.
I see that that might not be the best solution.
Please take note of me repeatedly stating in the text that the writer may use a different attribute naming scheme, file location etc, in their particular case if need be. I surmised that it would be enough to hedge against the possibility of sounding too prescriptive.  However, based on the comments that you and other reviewers have kindly provided, I see that I may have been wrong. Would you prefer just having the basic recommendation from the spreadsheet in the chapter for the time being? We can add other recommendations later, once we have reached a consensus of what said recommendations should be. Thoughs? Comments? @redhat-documentation/ccs-style-council

Reply:
Hi,
I can see @stoobie 's point-of-view about this content being more about implementation. However, I also get @inoxx03 's comment that product names and version attributes is a gray area (between style and implementation-FCC-asciidoc). When discussing product names and versions, I find it difficult to avoid discussing attributes. Let's include attributes and state the broadly accepted principles using them is a best practice that we encourage and support. The implementation details will work themselves out.